<a href="https://colab.research.google.com/github/Allajka/statistics/blob/main/seminar_7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import scipy.stats as stats

# Задание 1

Даны две независимые выборки. Не соблюдается условие нормальности:

**x1:** 380, 420, 290

**y1:** 140, 360, 200, 900

Сделайте вывод по результатам, полученным с помощью функции.

**Решение:**

Не соблюдается условие нормальности, используем непараметрические тест.

Так как две независимые выборки выбираем **критерий Манна- Уитни**.

In [2]:
x1 = np.array([380, 420, 290])
y2 = np.array([140, 360, 200, 900])

In [4]:
stats.mannwhitneyu(x1, y2)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

**Ответ:**

pvalue=0.62 **>** а = 0.05

При статистической значимости в а = 0,05(5%) принимаем нулевую гипотезу - статистически важных различий нет.

# Задача 2

Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. 

Есть ли статистически значимые различия?

**1е измерение до приема препарата:** 150, 160, 165, 145, 155

**2е измерение через 10 минут:** 140, 155, 150, 130, 135

**3е измерение через 30 минут:** 130, 130, 120, 130, 125

**Решение:**

Не соблюдается условие нормальности(или неизвестно), используем непараметрические тест.

Множественное сравнение зависимых данных, используем **критерий Фридмана**

In [5]:
before = np.array ([150, 160, 165, 145, 155])
after_10_min = np.array ([140, 155, 150, 130, 135])
after_30_min = np.array ([130, 130, 120, 130, 125])

In [6]:
stats.friedmanchisquare(before, after_10_min, after_30_min)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

**Ответ:**

pvalue=0.008 **<** а = 0.05

При статистической значимости в а = 0,05(5%) принимаем альтернативную гипотезу: есть статистически значимые различия.

# Задание 3

Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

**1е измерение до приема препарата:** 150, 160, 165, 145, 155

**2е измерение через 10 минут:** 140, 155, 150, 130, 135

Есть ли статистически значимые различия?

**Решение:**

Не соблюдается условие нормальности(или неизвестно), используем непараметрические тест.

Две зависимые выборки, используем **критерий Уилкоксона.**

In [8]:
x_before = np.array([150, 160, 165, 145, 155])
x_after_10_min = np.array([140, 155, 150, 130, 135])

In [10]:
stats.wilcoxon(x_before, x_after_10_min)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

**Ответ:**

pvalue=0.0625 **>** а = 0.05

При статистической значимости в а = 0,05(5%) принимаем нулевую гипотезу - статистически важных различий нет.

# Задание 4

Даны 3 группы учеников плавания. Время на дистанцию 50 м составляют

**Первая группа:** 56, 60, 62, 55, 71, 67, 59, 58, 64, 67

**Вторая группа:** 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

**Третья группа:** 57, 67, 49, 48, 47, 55, 66, 51, 54

Есть ли различия между группами?

**Решение:**

Не соблюдается условие нормальности(или неизвестно), используем непараметрические тест.

Множественное сравнение независимых данных, используем **критерий Крускала – Уоллиса(Краскела – Уоллиса).**

In [11]:
gr_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [12]:
stats.kruskal(gr_1, gr_2, gr_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

**Ответ:**

pvalue=0.065 **>** а = 0.05

При статистической значимости в а = 0,05(5%) принимаем нулевую гипотезу - статистически важных различий нет.

# Задание 5 

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

**Решение:**

Нормальное распределение, используем параметрические тест.

Будем использовать **t-критерий (критерий Стьюдента)**, т.к. 𝞼(среднеквадратическое отклонение) генеральной совокупности неизвестна.

* Формула: 

$$t = \frac{\overline{x} - μ} {\sqrt{\frac{D(X)}{n}}}$$

* Гипотеза:

  $$ H_0: μ_1 = μ_0 = 2.5 $$

  $$ H_1: μ_1 \neq μ _0$$

* Выбор уровня статистической значимости α(альфа):

  α(альфа) = 0,05(5%) - по условию



In [3]:
values = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]) 

In [4]:
dispersion = np.var(values, ddof=1) # Дисперсия
dispersion

0.02472888888888891

In [6]:
average = np.mean(values) # Среднее арифметиическое
average

2.5279999999999996

In [7]:
μ = 2.5
t = (average - μ) /np.sqrt(dispersion/len(values)) # t-критерий (критерий Стьюдента)
t

0.5630613661802959

Найдем табличное значение t-критерия для двусторонего теста а = 0,05/2 = 0,025

In [8]:
tt=stats.t.ppf(0.975,len(values)-1)
print(f'Табличное значение t-критерия: {tt: .4f}')

Табличное значение t-критерия:  2.2622


Или найдем табличное значение t-критерия а = 0,05

In [10]:
tt=stats.t.ppf(0.95,len(values)-1)
print(f'Табличное значение t-критерия: {tt: .4f}')

Табличное значение t-критерия:  1.8331


**Ответ: Принимаем нулевую гипотезу с уровенем статистической значимости 5%, что партия изготавливается со средним арифметическим 2,5 см.**